### ISSUE TITLE
[TW-1379] Color by status (waiting, pending, etc)

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1407]
_Will Adams on 2014-07-23T03:51:18Z says:_ 

 It seems like it should be possible to color tasks differently depending on status (pending/waiting/etc).

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2014-07-23T03:51:18Z
Modified: 2014-07-24T14:04:31Z
```

**Comment by taskwarrior** [comment]
_Renato Alves on 2014-07-23T17:25:56Z says:_ 

 Hi William,

In general pending, waiting and complete tasks don't mix with each other, unless you use the `all` report or the specific `waiting` and `completed` reports..
Other reports generally hide *complete* and *waiting* tasks.

Can you elaborate in which situation would this feature be visually helpful?

**Comment by taskwarrior** [comment]
_Will Adams on 2014-07-23T20:02:54Z says:_ 

 I'm using `depends:id` and `wait:date` extensively to keep most of my uncompleted tasks out of the reports I'm viewing when I'm actually working on them and deciding what to do next.  I like to be able to quickly input a lot of tasks throughout each day as issues arise without all of those tasks cluttering up my most frequently used reports and making it harder to see what actually needs attention.  

Basically, I'd like to be able to use `wait:date` and `wait:someday` similarly to how I'm currently using `depends:id`, which works great.  I can quickly use `+BLOCKED` on a report to completely hide blocked tasks, but when I view, for example, a report that shows all the tasks of a certain project, the blocked tasks are shown but are very easily distinguished from the others because of the different coloring.  

It'd be nice to include completed tasks with a color that stands out from the others on those kinds of reports as well (reports showing a single project or tag, or all tasks due withing a short time-frame, etc).

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2014-07-23T22:42:37Z says:_ 

 Hi Will,

I have good news.  This is already implemented in 2.4.0, and partly implemented in 2.3.0.  You can use color these color rules in 2.3.0, which make use of virtual tags:

```
color.tag.BLOCKED=...
color.tag.UNBLOCKED=...
color.tag.BLOCKING=...
color.tag.SCHEDULED=...
color.tag.UNTIL=...
color.tag.WAITING=...
color.tag.PARENT=...
color.tag.CHILD=...
```

In 2.4.0, there are new virtual tags to complete your request:

```
color.tag.PENDING=...
color.tag.COMPLETED=...
color.tag.DELETED=...
```

**Comment by taskwarrior** [comment]
_Will Adams on 2014-07-23T23:04:13Z says:_ 

 Good news indeed!  

Thank you,
Will
